phoenix-core/src/it/java/org/apache/phoenix/end2end/TableSnapshotReadsMapReduceIT.java

```diff
@@ -274,6 +293,9 @@ private void configureJob(Job job, String tableName, String inputQuery, String c
 
       assertFalse("Should only have stored" + result.size() + "rows in the table for the timestamp!", rs.next());
     } finally {
+      if (isSnapshotRestoreDoneExternally) {
+        assertRestoreDirCount(conf, tmpDir.toString(), 1);
```

Comment:
Could you assert the restoredircount is more than 1 if isSnapshotRestoreDoneExternally is set to false ?

Reply:
@shahrs87 There were two levels of subdirectories getting created for snapshot restore on every scan: 
https://github.com/apache/phoenix/blob/55f1362fc52eeabed139728dae153518883743a5/phoenix-core/src/main/java/org/apache/phoenix/mapreduce/util/PhoenixMapReduceUtil.java#L185

https://github.com/apache/phoenix/blob/55f1362fc52eeabed139728dae153518883743a5/phoenix-core/src/main/java/org/apache/phoenix/iterate/TableSnapshotResultIterator.java#L81

I have removed those in the original flow so now the directory gets cleaned up every single scan and gets created again for the next scan with the same directory structure. 

I can assert here no existence of the restore directory in the original flow.